### ISSUE TITLE
Can't select objects out of boundary

### ISSUE THREAD

**Issue by coldspark29** [xournalpp/xournalpp#1292]
- OS: probably any
 - Version: probably any with the new input system

If you past an object (like a picture) into the journal and it juts out of the boundary, you can't select or delete it anymore. The only way is to overwrite it. 

The new input system did solve the issue of drawing over the boundaries and it would be nice if the same would work for selecting.

**Comment by LittleHuba** [comment]
This should be possible if you use the object selection tool and click anywhere on the image.

**Comment by coldspark29** [comment]
Yes, but this also happened to me with a handwritten word close to the border today. In this case selecting every letter at once would be tedious. I can work with the object selection, now that I know it exists. I still think that it would be good to implement this, although it is not urgent.

**Comment by LittleHuba** [comment]
Enabling area selection outside of the page is not as easy as it sounds. It has one major problem. Xournal++ does currently not support selections over multiple pages in one go. Meaning enabling users to extend the selection area outside the active page would also allow them to extend it onto another page. But Xournal++ would simply ignore everything selected on the second page thus leading to very awkward behaviour that would definitely cause some ire.

So in essence we would need to go the full way and enable selections on multiple pages. But that is a really big feature request as we would not only need to rewrite big parts of the selection algorithms but also take the page layout into account.

You'll have to wait for that some time...

**Comment by coldspark29** [comment]
Okay, no worries. I can work with it using the select object tool for now. Thought it would be easier to implement.

**Comment by braupach** [comment]
> Enabling area selection outside of the page is not as easy as it sounds. It has one major problem. Xournal++ does currently not support selections over multiple pages in one go. Meaning enabling users to extend the selection area outside the active page would also allow them to extend it onto another page. But Xournal++ would simply ignore everything selected on the second page thus leading to very awkward behaviour that would definitely cause some ire.

While I see that this behaviour would be a bit awkward I'd say it is better than the current behaviour. The current behaviour is: 
- select a group of lines
- drag it [partially] over the left boundary (the right one isn't a problem as you can increase the page size)
- no way to select the lines as a group (object selection has the problem that you can only select single lines, destroying their alignment, or some of the lines being completly out of bounds)

Am I overlooking some feature which would allow me to select these lines?

This also wouldn't be _that_ akward as it is coherent with moving objects partially over the boundary but not onto different pages. And i'd say it is currently quite akward to not being able to select lines you just moved.

**Comment by zandivx** [comment]
An intuitive (partial) fix for this problem in my opinion would be to modify the rectangular selection tool to extend its containment area beyond the page borders if the rectangle touches the page border. I tried it this way: When the selection rectangle touches a page edge (within 1pt tolerance), its effective containment area extends to "infinity" (1e9) in that direction. This means if you drag your selection all the way to the page border, any part of an object that sticks out beyond the page is automatically considered "inside" the selection. If the selection doesn't reach the page edge, behavior is unchanged.
I did it only for rectangular selection for now. This does not open the can of multi-page selections (therefore partial fix).
The change is ~30 lines mostly in `RectangularSelector::contains`. Happy to open a PR if there's interest.
